corehq/apps/accounting/static/accounting/js/payment_method_handler.js

```diff
@@ -36,19 +36,20 @@ hqDefine('accounting/js/payment_method_handler', [
                 // the subscription summary info, use this hacky way to update
                 // the total of the general amount so the user is
                 // aware something happened.
-                var addedAmount = self.costItem().general_credit().addAmount(),
-                    $generalCredit = $('.js-general-credit');
+                var $generalCredit = $('.js-general-credit');
 
                 // make sure this only applies on pages where that class is present
-                if ($generalCredit) {
-                    var newValue = parseFloat($generalCredit.text().replace('USD ', '').trim()) + parseFloat(addedAmount);
+                if ($generalCredit && self.costItem() && self.costItem().general_credit) {
+
+                    var addedAmount = self.costItem().general_credit().addAmount(),
```

Comment:
should this say
```suggestion
                    var addedAmount = self.costItem().general_credit().addAmount(0),
```

Reply:
no...it's fetching the amount, not setting it